docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+ 
+# 5.0: UI Properties
+
+The UI component of the Z-JSON Instrument contains information on how we render our `fields` and `helpers` objects
+on the front-end. It is composed of JSON objects whose keys match those of the field or helper object it's describing.
+This component should only make decisions on which HTML or HTML Form element those objects get rendered as,
+and nothing else. It may also provide additional UI properties that assist in the rendering.
+
+The UI component object has the following general format:
+
+```js
+{
+    ...
+    "ui" : {
+        "nameOfFieldOrHelper" : {
+            "type": string,
+            "options": {
+                ...
+            },
+        },
+        ...
+    }
+}
+```
+
+A UI object type may or may not have type-specific `options` that configure the rendering of the front-end element.
+
+## 5.1 UI Types
+
+For fields:
+
+- [checkbox](#511-checkbox-element)
+- [date](#512-date-element)
+- [number](#513-numeric-element)
+- [radio](#514-radio-button-element)
+- [select](#515-select-element)
+- [text](#516-text-element)
+- [textarea](#517-textarea-element)
+- [time](#518-time-element)
+
+For helpers:
+
+- [header](#519-header-element) (from h1 to h6)
+- [link](#5110-link-element)
+- [static](#5111-static-element)
+
+These are all mappable to some LORIS React Form element, except for "header" which is native to HTML.
+
+### 5.1.1 Checkbox Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "checkbox"
+        }
+    }
+}
+```
+
+### 5.1.2 Date Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "date"
+        }
+    }
+}
+```
+
+### 5.1.3 Numeric Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "number"
+        }
+    }
+}
+```
+
+### 5.1.4 Radio Button Element
+
+```js
+{
+    "ui": {
+        ...
+        "nameOfRadioButtons" : {
+            "type" : "radio",
+            "options" : {
+                "vertical" : boolean
+            }
+        }
+    }
+}
+```
+
+`options.vertical`: Boolean. True or false that the radio group is rendered vertically. The default value is false, or
+horizontally. 
+
+### 5.1.5 Select Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "select"
+        }
+    }
+}
+```
+
+### 5.1.6 Text Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "text",
+            "options": {
+                "placeholder": string
+            }
+        }
+    }
+}
+```
+
+`options.placeholder`: String. Text that is rendered inside the text element as a placeholder before user input.
+
+### 5.1.7 Textarea Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "textarea",
+            "options": {
+                "placeholder": string
+            }
+        }
+    }
+}
+```
+
+`options.placeholder`: String. Text that is rendered inside the text element as a placeholder before user input.
```

Comment:
I don't see any reason to separate text and textarea, and not just make them options for the same type.

Reply:
this goes back to the comment on line 830

for me this was where we map input types to HTML types

Reply:
I think this should be mapping field types to UI concepts, not to HTML types. There are other ways than HTML to build a UI, different versions of HTML, and there are multiple ways even in HTML to build the same UI concept.